### DISCUSSION TITLE
Perform *only* security updates?

### DISCUSSION THREAD

**Discussion by zjullion** [renovatebot/renovate#15490] [Request Help] [answered]
My team is trying to evaluate `renovate` as our tool of choice for handling security vulnerabilities in `NodeJS` applications. We are using the GitHub app. Is it possible to configure `renovate` to perform *only* security updates? We've tried the following `renovate.json` (posting only relevant sections):

```
{
  "extends": ["config:base"],
  "packageRules": [
    {
      "enabled": false,
      "groupName": "everything",
      "matchManagers": ["npm"],
      "matchPackagePatterns": ["*"],
      "separateMajorMinor": false
    }
  ],
  "vulnerabilityAlerts": {
    "packageRules": [
      {
        "groupName": "everything",
        "matchManagers": ["npm"],
        "matchPackagePatterns": ["*"]
      }
    ]
  }
}
```

but this just disables everything. Is there a way to get a package / group of packages to be `enabled` for security updates only?

**Comment by zjullion** [comment, marked as answer]
Figured it out - you need `"enabled": true` under `vulnerabilityAlerts`, such as:

```
{
  "extends": ["config:base"],
  "packageRules": [
    {
      "enabled": false,
      "groupName": "everything",
      "matchManagers": ["npm"],
      "matchPackagePatterns": ["*"],
      "separateMajorMinor": false
    }
  ],
  "vulnerabilityAlerts": {
    "enabled": true
  }
}
```

**Reply by viceice** [reply]
please don't use nested package rule, you can probably remove it at all from vulnerability alerts object

**Reply by zjullion** [reply]
Edited!

**Comment by liepins101** [comment]
Also my team is trying to evaluate renovate as our tool of choice for handling security vulnerabilities in Java applications. The only difference is that we are using Renovate Runner project for our self-hosted GitLab. 

As far as I understand approach mentioned above can not be used by us because vulnerabilityAlerts configuration is supported only by GitHub platform. Is there any alternative?

We also considered to enable Dependency Dashboard and set dependencyDashboardApproval to true so in theory vulnerability remediation PRs would still get created immediately without requiring approval and assignee would be informed only in case of vulnerability remediation PRs. But once again this seems not to be a case for self-hosted GitLab

**Reply by viceice** [reply]
yes, it's only supported on GitHub. there is work in progress, search issues/ pull request

**Reply by weirdolucifer** [reply]
@viceice @liepins101 Is there any way possible to configure Renovate to perform only security updates for self-hosted GitLab now?

**Reply by viceice** [reply]
yes, checkout the osv vulnerability feature 

- https://github.com/renovatebot/renovate/discussions/20542

**Reply by weirdolucifer** [reply]
@viceice That's really cool feature. However, I am struggling to raise PR only on security updates.
`renovate.json` looks like this:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "gitLabAutomerge": true,
  "pip_requirements": {
    "fileMatch": ["[a-zA-Z-_]+\/[a-zA-Z-_]*requirements[a-zA-Z-_]*\\.txt$", "requirements.txt$"]
  },
  "packageRules": [
    {
      "description": "Trigger breaking release for major updates",
      "matchPackageNames": ["renovate/renovate"],
      "matchUpdateTypes": ["major"],
      "semanticCommitType": "feat",
      "commitMessageSuffix": "BREAKING CHANGE: Major update"
    },
    {
      "description": "Automatically merge minor updates",
      "matchPackageNames": ["renovate/renovate"],
      "matchUpdateTypes": ["minor", "patch", "digest"],
      "automerge": true,
      "automergeType": "branch",
      "requiredStatusChecks": null
    }
  ],
  "vulnerabilityAlerts": {
    "enabled": true
  },
  "osvVulnerabilityAlerts": true
}
```

Raised MR What to expect Section:

![Screenshot 2023-04-05 at 9 14 30 AM](https://user-images.githubusercontent.com/26038535/229987603-6dbec23a-f3c5-4d7a-a042-be21bbfe4214.png)

After merging this Onboarding PR, It will generate PR for all packages. I want PR only for [SECURITY] tags, what configuration do I need to update?

**Reply by viceice** [reply]
you need a disable all package rule first

**Reply by weirdolucifer** [reply]
I have disabled the `packageRules`. But, the same results are there, Is there any documentation for raising PR only for security vulnerabilities?

**Reply by viceice** [reply]
not disable your rules, add a rule as first which matches all packages and set `enabled=false`

**Reply by weirdolucifer** [reply]
Thanks, for support. working now.

**Reply by viceice** [reply]
please provide the solution as separate answer, so we can mark it as answer for future readers

**Reply by weirdolucifer** [reply]
@viceice, I am currently working with both Mend's "app.whitesourcesoftware.com" and the "Mend renovate BOT". I have noticed that there are differences in the suggested vulnerabilities between the two tools, which may be due to their use of different databases. Specifically, the Mend BOT uses the OSV database, while "app.whitesourcesoftware.com" uses a different database. 

For instance, according to "app.whitesourcesoftware.com", there is a vulnerability in Uglify 3.4.10, but no such vulnerability is reported in OSV.dev. 

<img width="1130" alt="Screenshot 2023-04-11 at 4 33 37 PM" src="https://user-images.githubusercontent.com/26038535/231167321-a3fd7750-0100-4822-8e0f-af2e5e0d1760.png">

<img width="1429" alt="Screenshot 2023-04-11 at 4 37 02 PM" src="https://user-images.githubusercontent.com/26038535/231167333-83d94c22-6367-437c-96bf-53a16e9ff878.png">


I am integrating both tools in the following way: first, the BOT will send a pull request based on any security vulnerabilities it detects. Then, after merging changes, the CI pipeline will run, and in a later stage, I will scan the project using "app.whitesourcesoftware.com", which may suggest the presence of additional vulnerabilities. Is there any way to overcome this problem and ensure that both tools are identifying all relevant vulnerabilities?

**Reply by viceice** [reply]
best option is to suggest an update to the osv database to include the missing vulnerability.

@rarkins any other things we can do here?

**Reply by rarkins** [reply]
> Is there any way to overcome this problem and ensure that both tools are identifying all relevant vulnerabilities?

No, they are two different databases and will remain so.

If you think Mend's CVE is incorrect and you are a customer then raise a support ticket so that the team can correct and try to work out how to avoid such mistakes in future (this repo is not the place).

If the OSV database is incorrect then either:
- Accept that, because it's a free DB, or
- Raise an issue or PR to the project, linking back to the original source (e.g. NVD)

**Reply by rarkins** [reply]
I took a quick look at that CVE and it appears that it was disputed, so security tools have begun removing it. I've sent some colleagues a message about it but I recommend you also raise a customer ticket if it's important to you.

**Reply by weirdolucifer** [reply]
Thanks @rarkins for the clarification, I'll discuss this with my team and raise a customer ticket. @viceice Is there any way to use Mend's CVE rather than osvVulnerabilityAlerts in renovate BOT to perform only security updates?

**Reply by rarkins** [reply]
@weirdolucifer it is not possible to use Renovate with Mend's DB like you are asking. However as a customer, you should be using Mend's "Repository Integrations" which do combine Renovate plus Mend's DB into a product called "Mend Remediate".

**Reply by rarkins** [reply]
This original post has been answered and resolved, so I would like to lock it and move on before it get too much more off-topic.

**Comment by weirdolucifer** [comment]
To perform only security updates for self-hosted GitLab, `renovate.json` looks like this:

```
{
  "extends": [ "config:base"],
  "packageRules": [
    {
      "enabled": false,
      "groupName": "everything",
      "matchPackagePatterns": ["*"],
      "separateMajorMinor": false
    }
  ],
  "vulnerabilityAlerts": {
    "enabled": true
  },
  "osvVulnerabilityAlerts": true
}
```

**Reply by viceice** [reply]
you only need this:
```json
{
  "extends": [ "config:base"],
  "packageRules": [
    {
      "enabled": false,
      "matchPackagePatterns": ["*"]
    }
  ],
  "vulnerabilityAlerts": {
    "enabled": true
  },
  "osvVulnerabilityAlerts": true
}
```
